python/dolfinx/wrappers/la.cpp

```diff
@@ -189,7 +190,10 @@ void la(py::module& m)
              self.insert(xtl::span(rows.data(), rows.size()),
                          xtl::span(cols.data(), cols.size()));
            })
-      .def("insert_diagonal", &dolfinx::la::SparsityPattern::insert_diagonal)
+      .def("insert_diagonal",
+           [](dolfinx::la::SparsityPattern& self,
+              const py::array_t<std::int32_t, py::array::c_style>& rows)
+           { self.insert_diagonal(xtl::span(rows.data(), rows.size())); })
```

Comment:
I don't think that `xtl::span` is required here. Usually an implicit construction works.